lib/server.js

```diff
@@ -255,6 +257,20 @@ class Server {
       throw new Error("Can't determine port before the server is started");
     }
 
+    return this._httpServer.address().port;
+  }
+  /**
+   * Gets the hostname that the server is requesting. The server must be started
+   * before this method is called.
+   * @function
+   * @name Server#port
+   * @return {string} The hostname
+   */
+  hostname() {
```

Comment:
This method doesn't work. It gets overwritten by a string in both the constructor and the `start` method. If it wasn't overwritten, it would return the port instead of the hostname. Please add test coverage and fix those problems.